kernel/xpmem_attach.c

```diff
@@ -162,6 +162,120 @@ xpmem_close_handler(struct vm_area_struct *vma)
 #endif
 }
 
+/* Find the range of non present entries */
+static u64
+xpmem_fault_vaddr_end(u64 vm_end, u64 addr, unsigned long max_pages)
+{
+	for (; max_pages; max_pages--, addr += PAGE_SIZE) {
+		if (addr >= vm_end) {
+			break;
+		}
+		if (xpmem_vaddr_to_PFN(current->mm, addr)) {
```

Comment:
since it's just checking if page is present, can we use xpmem_vaddr_to_pte_offset?
same for xpmem_fault_vaddr_start

Reply:
fixed for the two cases